observability/micrometer/src/main/java/org/wildfly/extension/micrometer/prometheus/PrometheusRegistryDefinition.java

```diff
@@ -0,0 +1,72 @@
+package org.wildfly.extension.micrometer.prometheus;
+
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.ReloadRequiredRemoveStepHandler;
+import org.jboss.as.controller.ReloadRequiredWriteAttributeHandler;
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.SimpleResourceDefinition;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.wildfly.extension.micrometer.MicrometerConfigurationConstants;
+import org.wildfly.extension.micrometer.MicrometerExtension;
+import org.wildfly.extension.micrometer.registry.WildFlyCompositeRegistry;
+
+public class PrometheusRegistryDefinition extends SimpleResourceDefinition {
+    public static final PathElement PATH_ELEMENT = PathElement.pathElement("registry", "prometheus");
+
+    static final RuntimeCapability<Void> MICROMETER_PROMETHEUS_CONTEXT_CAPABILITY =
+            RuntimeCapability.Builder.of("org.wildfly.extension.micrometer.prometheus.http-context",
+                            PrometheusContextService.class)
```

Comment:
Capability service types should _never_ leak the MSC API.
It seems to me that this capability either should not have any service type (if there are no services that would consume it) or use a service type of Void.